test/e2e/create_test.go

```diff
@@ -173,12 +173,55 @@ func (tc *testContext) testMachineConfiguration(t *testing.T) {
 		// Replace the known private key with a randomly generated one.
 		err = tc.createPrivateKeySecret(false)
 		require.NoError(t, err, "error replacing private key secret")
+		// Wait for machines to be reconciled by deletion
+		err = tc.waitForMachinesDeleted(machines.Items)
+		require.NoError(t, err, "error waiting for machines deletion after private key secret change")
+		// Wait machines to be provisioned and update the machines list
+		machines, err = tc.waitForWindowsMachines(int(gc.numberOfMachineNodes), "Provisioned", false)
+		require.NoError(t, err, "error waiting for Windows Machines to be provisioned")
 	}
 	err = tc.waitForConfiguredWindowsNodes(gc.numberOfMachineNodes, false, false)
 	assert.NoError(t, err, "Windows node creation failed")
```

Comment:
Won't this suffice as a hack?:
```suggestion
	if tc.CloudProvider.GetType() == config.AzurePlatformType {
		// Wait machines to be provisioned and update the machines list
		machines, err = tc.waitForWindowsMachines(int(gc.numberOfMachineNodes), "Provisioned", false)
		require.NoError(t, err, "error waiting for Windows Machines to be provisioned")
	}
```

Reply:
No, at this point, all machines are in `Running` state, hence querying for `Provisioned` will return an empty result set leaving the log collection as a no-op.

In that sense, I did explore a [hack](https://github.com/openshift/windows-machine-config-operator/compare/8c16267d07ba12af9ad92ff25a32b9a5f42d5bcc..ba94db7592e39f2d7d1103bd06ad653fa6262701) that only refreshes the list of machines and later added the wait to tackle the timing issue in appearing in https://github.com/openshift/windows-machine-config-operator/pull/1770

Reply:
Did you hack not suffice? Though it still does not answer how those Machines are getting deleted.

Reply:
> Did you hack not suffice? 

No, without waiting for all the outdated machines to be deleted, there is a chance to pick up of of them while refreshing the list of machines.

Reply:
> Though it still does not answer how those Machines are getting deleted.

See  https://github.com/openshift/windows-machine-config-operator/pull/1793#discussion_r1327638674

Reply:
I think it might be simpler to just do:
```suggestion
	if tc.CloudProvider.GetType() == config.AzurePlatformType {
		// Update the machines list
		machines, err = tc.waitForWindowsMachines(int(gc.numberOfMachineNodes), "Running", false)
		require.NoError(t, err, "error waiting for Windows Machines to be running")
	}
```
which is slightly different from your hack. Or am I missing something that needs the tests to wait for the Machine to be deleted.

Reply:
Addressed in [cd47d34 ](https://github.com/openshift/windows-machine-config-operator/commit/cd47d34b0ae6a114ee3a752c1bfdca380fc3d156)